### PR TITLE
Core logic for processing orders and positions complete

### PR DESCRIPTION
**Pull request by speckospock**
Still need to create unit tests for everything, but it works as intended!

### CODE REVIEW

File: src/db/index.js [hunk 1]
```diff
@@ -110,56 +118,229 @@ const topSells = callback => {
...
+      if (result.dataValues.type === type) {
+        console.log(result.dataValues.orders);
+        let newInfo = result.dataValues.orders.reduce((memo, el) => {
+          memo.priceSum += el.price;
+          memo.volSum += el.volume;
+          return memo;
+        }, { priceSum: price, volSum: volume });
+        console.log(newInfo);
+        // console.log('would write ', [...result.dataValues.orders, { price, volume }]);
+        console.log('hi hi', result.dataValues);
```
Comment (kennyxcao) [inline src/db/index.js]: some redundant console logging

File: src/db/index.js [hunk 2]
```diff
@@ -110,56 +118,229 @@ const topSells = callback => {
...
+        console.log('calculated price: ', (newInfo.priceSum / (result.dataValues.orders.length + 1)));
+        console.log('calculated volume: ', newInfo.volSum);
+
+        Position.update({
+          price: (newInfo.priceSum / (result.dataValues.orders.length + 1)).toFixed(4),
+          volume: newInfo.volSum,
+          orders: [...result.dataValues.orders, { price, volume }],
+        }, {
+          where: { userId },
+        }).then(res => console.log(res));
```
Comment (kennyxcao) [inline src/db/index.js]: should have error catch handlers for promises

File: src/db/index.js [hunk 3]
```diff
@@ -110,56 +118,229 @@ const topSells = callback => {
...
+              price,
+              type,
+              volume: newVolume,
+              orders: [{ price, volume: newVolume }],
+            });
+          //if no remainder, simply destroy the position
+          } else {
+            result.destroy();
+          }
+        } else {
```
Comment (kennyxcao) [inline src/db/index.js:191]: Awesome implementation of a complex logic

File: src/db/index.js [hunk 4]
```diff
@@ -110,56 +118,229 @@ const topSells = callback => {
...
+          let orders = [...result.dataValues.orders];
+          console.log('orders: ', orders);
+          for (let vol = 0; vol < volume; vol) {
```
Comment (kennyxcao) [inline src/db/index.js]: while loop is probably better here

**Review by kennyxcao — CHANGES_REQUESTED** [review]

File: src/db/index.js (hunk 4 above)
Comment (speckospock) [reply src/db/index.js]: Made requested change

**Pull request merged**